traversal/structure/Structure.java

```diff
@@ -107,24 +110,36 @@ private void recordEdge(StructureEdge<?, ?> edge) {
             edge.to().in(edge);
         }
     }
-    public List<Structure> asGraphs() {
-        if (structures == null) {
-            structures = new ArrayList<>();
-            Set<StructureVertex<?>> verticesToVisit = new HashSet<>(this.vertices.values());
-            Set<StructureEdge<?, ?>> edgesToVisit = new HashSet<>(this.edges);
-            while (!verticesToVisit.isEmpty()) {
-                Structure newStructure = new Structure();
-                splitGraph(verticesToVisit.iterator().next(), newStructure, verticesToVisit, edgesToVisit);
-                if (newStructure.vertices().size() > 1 || newStructure.vertices().iterator().next().id().isRetrievable()) {
-                    structures.add(newStructure);
-                }
+
+    public List<Structure> splitConnected() {
+        return splitConnected(emptyList());
+    }
+
+    public List<Structure> splitConnected(List<? extends Identifier> forceConnect) {
```

Comment:
when we split a structure into subgraphs, we can mandate that some vertices are in the same subgraph, even if they have no direct edge connections